extension-impl/lookout/src/main/java/com/alipay/sofa/rpc/event/LookoutSubscriber.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.rpc.event;
+
+import com.alipay.sofa.rpc.common.RemotingConstants;
+import com.alipay.sofa.rpc.common.RpcConfigs;
+import com.alipay.sofa.rpc.common.RpcConstants;
+import com.alipay.sofa.rpc.common.RpcOptions;
+import com.alipay.sofa.rpc.context.RpcInternalContext;
+import com.alipay.sofa.rpc.core.request.SofaRequest;
+import com.alipay.sofa.rpc.core.response.SofaResponse;
+import com.alipay.sofa.rpc.lookout.RpcLookout;
+import com.alipay.sofa.rpc.model.RpcClientLookoutModel;
+import com.alipay.sofa.rpc.model.RpcServerLookoutModel;
+
+/**
+ * Collect the raw information for lookout by listening to events.
+ *
+ * @author <a href="mailto:lw111072@antfin.com">LiWei.Liengen</a>
+ * @version $Id: MetricsSubscriber.java, v 0.1 2018年04月23日 下午9:48 LiWei.Liengen Exp $
+ */
+public class LookoutSubscriber extends Subscriber {
```

Comment:
if we think Lookout is not essential sometime, maybe we should collect asynchonized?

Reply:
it is asynchonized . can see `com.alipay.sofa.rpc.event.LookoutSubscriber#LookoutSubscriber`